RevenueCatUI/CustomerCenter/ViewModels/ManageSubscriptionsViewModel.swift

```diff
@@ -0,0 +1,185 @@
+            price: subscribedProduct.localizedPriceString,
+            nextRenewalString: currentEntitlement.expirationDate.map { dateFormatter.string(from: $0) } ?? nil,
+            willRenew: currentEntitlement.willRenew,
+            productIdentifier: subscribedProductID,
+            active: currentEntitlement.isActive
+        )
+    }
+
+    func loadCustomerCenterConfig() async {
+        self.configuration = CustomerCenterConfigTestData.customerCenterData
+    }
+
+    #if os(iOS) || targetEnvironment(macCatalyst)
+    func handleAction(for path: CustomerCenterConfigData.HelpPath) {
+        switch path.type {
+        case .missingPurchase:
+            self.showRestoreAlert = true
+        case .refundRequest:
+            Task {
+                guard let subscriptionInformation = self.subscriptionInformation else { return }
+                let productId = subscriptionInformation.productIdentifier
+                let status = try await purchasesProvider.beginRefundRequest(forProduct: productId)
+                switch status {
+                case .error:
+                    self.refundRequestStatus = "Error when requesting refund, try again"
+                case .success:
+                    self.refundRequestStatus = "Refund granted successfully!"
+                case .userCancelled:
+                    self.refundRequestStatus = "Refund canceled"
+                }
+            }
+        case .changePlans, .cancel:
+            Task {
+                try await purchasesProvider.showManageSubscriptions()
+            }
+        default:
+            break
+        }
+    }
+    #endif
+
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, *)
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+private final class ManageSubscriptionPurchases: ManageSubscriptionsPurchaseType {
+
+    func beginRefundRequest(forProduct productID: String) async throws -> RevenueCat.RefundRequestStatus {
+        try await Purchases.shared.beginRefundRequest(forProduct: productID)
+    }
+
+    func showManageSubscriptions() async throws {
+        try await Purchases.shared.showManageSubscriptions()
+    }
+
+    func customerInfo() async throws -> RevenueCat.CustomerInfo {
+        return try await Purchases.shared.customerInfo()
+    }
+
+    func products(_ productIdentifiers: [String]) async -> [StoreProduct] {
+        return await Purchases.shared.products(productIdentifiers)
+    }
+
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+private extension SubscriptionPeriod {
+
+    var durationTitle: String {
+        switch self.unit {
+        case .day: return "day"
+        case .week: return "week"
+        case .month: return "month"
+        case .year: return "year"
+        default: return "Unknown"
```

Comment:
Same for these, probably need to be localized? Separately to avoid having to localize all the combinations I think.